sql/privileges.go

```diff
@@ -19,16 +19,20 @@ type PrivilegedOperation struct {
 	Database          string
 	Table             string
 	Column            string
+	Routine           string
+	IsProcedure       bool // true if the routine is a procedure, false if it's a function
```

Comment:
is this used? clicking through the uses it seems like procedures self-select into the admin check

Reply:
good point. I'm going to implement function perm which is a gap, and see if I can drop this.

Reply:
It is used in UserHasPrivileges.